DMS/app/src/main/java/com/example/kbb12/dms/Model/InsulinTakenRecord/InsulinTakenDatabase.java

```diff
@@ -48,15 +30,40 @@ public void addEntry(int day,int month,int year,int hour,int minute,Double dose,
         }else{
             trueOrFalse="FALSE";
         }
-        write.execSQL("INSERT INTO " + InsulinTakenContract.ContentsDefinition.TABLE_NAME + " (" + InsulinTakenContract.ContentsDefinition.COLUMN_ONE_TITLE + ", " + InsulinTakenContract.ContentsDefinition.COLUMN_TWO_TITLE + ", " + InsulinTakenContract.ContentsDefinition.COLUMN_THREE_TITLE + ", " + InsulinTakenContract.ContentsDefinition.COLUMN_FOUR_TITLE + ")VALUES(\"" + formatDate(day,month,year) + "\", \"" + formatTime(hour,minute) + "\", '" + trueOrFalse + "', " + dose + ")");
+        write.execSQL("INSERT INTO " + InsulinTakenContract.ContentsDefinition.TABLE_NAME + " (" + InsulinTakenContract.ContentsDefinition.COLUMN_ONE_TITLE + ", " + InsulinTakenContract.ContentsDefinition.COLUMN_TWO_TITLE + ", " + InsulinTakenContract.ContentsDefinition.COLUMN_THREE_TITLE + ")VALUES(\"" + formatDateTime(year,month,day,hour,minute) + "\", '" + trueOrFalse + "', " + dose + ")");
+    }
+
+    @Override
+    public IInsulinTakenEntry getMostRecentBolus() {
+        /*
+        Get the entry with the maximum time which isn't Basal insulin.
+         */
+        Cursor cursor = write.rawQuery("Select MAX("
+                +InsulinTakenContract.ContentsDefinition.COLUMN_ONE_TITLE+"),"+
+                InsulinTakenContract.ContentsDefinition.COLUMN_TWO_TITLE+", "+
+                InsulinTakenContract.ContentsDefinition.COLUMN_THREE_TITLE+
+                " from "+InsulinTakenContract.ContentsDefinition.TABLE_NAME+
+                " where "+InsulinTakenContract.ContentsDefinition.COLUMN_TWO_TITLE+"=?",
+                new String[]{"FALSE"});
+        if(!cursor.moveToNext()||cursor.getString(0)==null) {
+            return null;
+        }
+        return new InsulinTakenEntry(parseTime(cursor.getString(0)),
```

Comment:
Try using cursor.getColumnIndex() here instead of a constant number, just to make sure no errors happen in the future